catalyst/dl/callbacks/confusion_matrix.py

```diff
@@ -94,13 +96,21 @@ def on_loader_end(self, state: State):
             self.class_names or \
             [str(i) for i in range(self.num_classes)]
         confusion_matrix = self._compute_confusion_matrix()
-        tb_callback = state.callbacks[self.tensorboard_callback_name]
-        self._plot_confusion_matrix(
-            logger=tb_callback.loggers[state.loader_name],
-            epoch=state.global_epoch,
-            confusion_matrix=confusion_matrix,
-            class_names=class_names,
-        )
+
+        if utils.get_rank() >= 0:
+            confusion_matrix = torch.from_numpy(confusion_matrix)
+            confusion_matrix = confusion_matrix.to(utils.get_device())
+            torch.distributed.reduce(confusion_matrix, 0)
+            confusion_matrix = confusion_matrix.cpu().numpy()
+
+        if utils.get_rank() <= 0:
```

Comment:
could you please use `state.distributed_rank` or `state.is_distributed_worker` instead?

Reply:
done